src/snapshotcloneserver/clone/clone_service_manager.cpp

```diff
@@ -248,9 +245,9 @@ int CloneServiceManager::Flatten(
         }
     }
 
-    DLockGuard dlockGuard(dlock_);
+    auto cloneInfoMetric = std::make_shared<CloneInfoMetric>(taskId);
```

Comment:
why not release dlock?
where cloneInfoMetric used?

Reply:
> why not release dlock?
> where cloneInfoMetric used?

dlock acquired before clonetask run and release after task finished. Here just init dlock.
cloneInfoMetric is some code before and used in CloneTaskInfo, here is just adjusted its position.